examples/cmd/logtransport.go

```diff
@@ -22,15 +21,23 @@ type logTransport struct {
 func (t *logTransport) RoundTrip(r *http.Request) (*http.Response, error) {
 	t0 := time.Now()
 
-	reqBody, _ := ioutil.ReadAll(r.Body)
-	r.Body = ioutil.NopCloser(bytes.NewBuffer(reqBody))
-
+	// Since the underlaying Transport.RoundTrip might consume the Request, and it's also needed later to log the process,
+	// it is needed to make an snapshot of the Request.Body in advance; otherwise, its will be no longer accessible.
+	bakupReqBody := utils.BackupRequestBody(r)
 	resp, err := t.T.RoundTrip(r)
+	r.Body = bakupReqBody
+
+	// Since the outter Transport.RoundTrip calling this one might also consume the Request, and it is also needed here to log
+	// the process, it will be done over a snapshot of the Request.Body; otherwise, its will be no longer accessible by outer Transports.
+	reqBody, _ := utils.ReadRequestAndRestore(r)
```

Comment:
why `_` here and the check for the error below?

Reply:
8b3d61ccdca52bf6a12a21bb4812c2f4b9b4bc8a thanks